src/Page/Form.purs

```diff
@@ -58,91 +76,39 @@ component =
     { form } <- H.get
     case form of
-      Success { expr } -> eval (EvalForm (setPage key val expr) a)
+      Success { expr } ->
+        eval (EvalForm (Lynx.Data.Form.setValue key val expr) a)
       _ -> pure a
 
-  evalPage :: (Key -> Maybe ExprType) -> Page Expr -> Either EvalError (Page ExprType)
-  evalPage get page = do
-    contents <- traverse (evalSection get) page.contents
-    pure page { contents = contents }
-
-  evalSection :: (Key -> Maybe ExprType) -> Section Expr -> Either EvalError (Section ExprType)
-  evalSection get section = do
-    contents <- traverse (evalField get) section.contents
-    pure section { contents = contents }
-
-  evalField :: (Key -> Maybe ExprType) -> Field Expr -> Either EvalError (Field ExprType)
-  evalField get field = do
-    description <- evalExpr get field.description
-    input <- evalInput get field.input
-    name <- evalExpr get field.name
-    visibility <- evalExpr get field.visibility
-    pure { description, key: field.key, input, name, visibility }
-
-  evalInput :: (Key -> Maybe ExprType) -> Input Expr -> Either EvalError (Input ExprType)
-  evalInput get = case _ of
-    Text input -> do
-      default <- traverse (evalExpr get) input.default
-      maxLength <- traverse (evalExpr get) input.maxLength
-      minLength <- traverse (evalExpr get) input.minLength
-      placeholder <- evalExpr get input.placeholder
-      required <- evalExpr get input.required
-      value <- traverse (evalExpr get) input.value
-      pure
-        ( Text
-          { default
-          , maxLength
-          , minLength
-          , placeholder
-          , required
-          , value
-          }
-        )
-    Toggle input -> do
-      default <- traverse (evalExpr get) input.default
-      value <- traverse (evalExpr get) input.value
-      pure (Toggle { default, value })
-
-  keysPage :: Page Expr -> Map Key ExprType
-  keysPage page = foldMap keysSection page.contents
-
-  keysSection :: Section Expr -> Map Key ExprType
-  keysSection section = foldMap keysField section.contents
-
-  keysField :: Field Expr -> Map Key ExprType
-  keysField field = case field.input of
-    Text { value: UserInput expr }
-      | Right value <- evalExpr (const Nothing) expr ->
-        Data.Map.singleton field.key value
-    Text { default: Just expr }
-      | Right value <- evalExpr (const Nothing) expr ->
-        Data.Map.singleton field.key value
-    Toggle { value: UserInput expr  }
-      | Right value <- evalExpr (const Nothing) expr ->
-        Data.Map.singleton field.key value
-    Toggle { default: Just expr }
-      | Right value <- evalExpr (const Nothing) expr ->
-        Data.Map.singleton field.key value
-    _ -> mempty
-
-  render :: State -> H.ComponentHTML Query
+  eval (DropdownQuery key message a) = case message of
+    Dropdown.Emit x -> a <$ eval x
+    Dropdown.Selected val -> eval (UpdateKey key val a)
+    Dropdown.VisibilityChanged visibility' -> do
```

Comment:
I think we want to ignore this query. Visibility on a dropdown is different from visibility on a field. This query let's us know when the dropdown containers visibility has been toggled on or off (so if a user clicks on the dropdown, the container's visibility will toggle `On` and the options will be shown, then if an option is selected of the user clicks elsewhere, the visibility will toggle `Off` and the container will hide.

Separate from that is our notion of `visibility` on a field, which is an expression to determine whether the field itself should be rendered or not.

Reply:
Ah! That makes sense. Thanks for the explanation.